src/Nancy/Owin/NancyMiddleware.cs

```diff
@@ -80,7 +80,9 @@ public static MidFunc UseNancy(NancyOptions options = null)
 
                         var url = CreateUrl(owinRequestHost, owinRequestScheme, owinRequestPathBase, owinRequestPath, owinRequestQueryString);
 
-                        var nancyRequestStream = new RequestStream(owinRequestBody, ExpectedLength(owinRequestHeaders), StaticConfiguration.DisableRequestStreamSwitching ?? false);
+                        var expectedLength = ExpectedLength(owinRequestHeaders);
+                        // If length is 0 just use empty memory stream; as there is no body
+                        var nancyRequestStream = (expectedLength == 0) ? (Stream)new MemoryStream() : new RequestStream(owinRequestBody, ExpectedLength(owinRequestHeaders) ?? 0, StaticConfiguration.DisableRequestStreamSwitching ?? false);

```

Comment:
We should probably use `expectedLength` instead of calling `ExpectedLength(owinRequestHeaders)` again?

Reply:
Lol; yes

Reply:
Changed